litmus-portal/authentication/pkg/entities/user.go

```diff
@@ -0,0 +1,67 @@
+package entities
+
+import (
+	"litmus/litmus-portal/authentication/pkg/utils"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+//User contains the user information
+type User struct {
+	ID        primitive.ObjectID `bson:"_id,omitempty" json:"_id,omitempty"`
+	UserName  string             `bson:"username,omitempty" json:"username,omitempty"`
```

Comment:
Should we use omitempty with ID, UserName, Role and Password? These fields will/should never be nil :thinking:

Reply:
- We do not want to send Password while sending responses for `/users` GET route, hence omitempty is required.
- I agree ID, UserName, Role are never be nil and is guaranteed from the server's side as there are validations and checks.
- but `omitempty` is significant only when we are de-serialising the json/bson. Hence I think there is no difference if we write or do not write `omitempty`

Reply:
IMO. It's better to remove `omitempty` if the fields are compulsory.